cmd/cmd.go

```diff
@@ -66,17 +65,36 @@ func ExecInCoreSnap() {
 
 	full := filepath.Join(newCore, exe)
 	if !osutil.FileExists(full) {
-		if rev, err := os.Readlink(oldCore); err != nil {
-			return
-		} else if revno, err := strconv.Atoi(rev); err != nil || revno < minOldRevno {
+		full = filepath.Join(oldCore, exe)
+		if !osutil.FileExists(full) {
 			return
 		}
+	}
 
-		full = filepath.Join(oldCore, exe)
-		if !osutil.FileExists(full) {
+	// ensure we do not re-exec into an older version of snapd, look
+	// for info file and ignore version of core that do not yet have
+	// it
+	fullInfo := filepath.Join(newCore, "/usr/lib/snapd/info")
+	if !osutil.FileExists(fullInfo) {
+		fullInfo = filepath.Join(oldCore, "/usr/lib/snapd/info")
+		if !osutil.FileExists(fullInfo) {
 			return
 		}
 	}
+	content, err := ioutil.ReadFile(fullInfo)
+	if err != nil {
+		logger.Noticef("cannot read info file %q: %s", fullInfo, err)
+		return
+	}
+	ver := regexp.MustCompile("(?ms)^VERSION=(.*)\n").FindStringSubmatch(string(content))
```

Comment:
You want `m`, sure, so the `^` and `$` match in any line of the file, but not `s`: you don't want that `.*` to eat everything to the end of the file.
Also, instead of `\n`, just use `$` (so that you don't *need* to have a newline at EOF).